src/lib/__tests__/RoutingManager-test.ts

```diff
@@ -537,6 +539,51 @@ describe('RoutingManager', () => {
         },
       });
     });
+    test('cleans URL on dispose', async () => {
+      const searchClient = createSearchClient();
+      const writeDelay = 400;
+
+      const search = instantsearch({
+        indexName: 'indexName',
+        searchClient,
+        routing: {
+          router: historyRouter({
+            writeDelay,
+          }),
+          stateMapping: simple(),
+        },
+      });
+
+      const fakeSearchBox: any = createFakeSearchBox();
+      search.addWidgets([fakeSearchBox]);
+
+      search.start();
+
+      await runAllMicroTasks();
+
+      // Wait for ${writeDelay} to pass then check URL has been initialized
+      await wait(1.5 * writeDelay);
+      expect(window.location.search).toEqual(``);
+      expect(window.history.length).toEqual(1);
+
+      // Trigger an update - push a change
+      fakeSearchBox.refine('Apple');
+
+      // Wait for ${writeDelay} to pass then check URL has been updated
+      await wait(1.5 * writeDelay);
+      expect(window.location.search).toEqual(
+        `?${encodeURI('indexName[query]=Apple')}`
+      );
+      expect(window.history.length).toEqual(2);
+
+      // Trigger a dispose
+      search.dispose();
```

Comment:
you're not testing here if the widgets get removed before the search

Reply:
To me I do because that's what `search.dispose()` does at first (remove widgets).
Is this the flow you have in mind?

```ts
...
const fakeSearchBox: any = createFakeSearchBox();
search.addWidgets([fakeSearchBox]);

search.start();
...
fakeSearchBox.refine('Apple');
... // check history
search.removeWidgets([fakeSearchBox])
... // check history
search.dispose()
.. // check history

```

Reply:
I know it's in Vue InstantSearch / Angular InstantSearch, but it would be interesting to see how that behaves, what actually gets unmounted when

Reply:
This test is supposed to be close to what happens in [Angular IS](https://github.com/algolia/angular-instantsearch/blob/master/src/instantsearch/instantsearch.ts#L247) and [Vue IS](https://github.com/algolia/vue-instantsearch/blob/513017a47d38ccddd0557c01ce732e6e5e7648d6/src/util/createInstantSearchComponent.js#L71-L78) when a route unmounts. 

Maybe we can try to be even closer to what's happening in Angular IS and Vue IS by simulating the cascading unmounting that occurs on route change. I'll do some research and update here.